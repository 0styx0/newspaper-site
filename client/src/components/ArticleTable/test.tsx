import * as React from 'react';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import * as casual from 'casual';
import ArticleLink from './Link';
import snapData from './__snapshots__/props.example';

describe('<ArticleLink>', () => {

    describe('snapshots', () => {

        it('renders correctly when url does NOT need to be decoded for viewing', () => {

            const tree = renderer.create(
                <MemoryRouter>
                    <ArticleLink url={snapData.oneWord.url} issue={snapData.oneWord.issue} />
                </MemoryRouter>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('renders correctly when url DOES need to be decoded for viewing', () => {

            const tree = renderer.create(
                <MemoryRouter>
                    <ArticleLink url={snapData.urlEncoded.url} issue={snapData.urlEncoded.issue} />
                </MemoryRouter>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
