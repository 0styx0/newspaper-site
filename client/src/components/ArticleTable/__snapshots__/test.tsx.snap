// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleLink> snapshots renders correctly when url DOES need to be decoded for viewing 1`] = `
<a
  href="/issue/16/story/dolorem%20voluptatem"
  onClick={[Function]}
>
  dolorem voluptatem
</a>
`;

exports[`<ArticleLink> snapshots renders correctly when url does NOT need to be decoded for viewing 1`] = `
<a
  href="/issue/35/story/suscipit"
  onClick={[Function]}
>
  suscipit
</a>
`;
